WireAPI/Tests/WireAPITests/Helpers/MockAPIServiceProtocol+Helpers.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+@testable import WireAPI
+@testable import WireAPISupport
+
+extension MockAPIServiceProtocol {
+
+    typealias Response = (statusCode: HTTPStatusCode, resourceName: String)
+
+    static func withResponses(_ responses: [Response]) -> MockAPIServiceProtocol {
+        let apiService = MockAPIServiceProtocol()
+        var responses = responses
+
+        apiService.executeRequestRequiringAccessToken_MockMethod = { request, _ in
+            guard !responses.isEmpty else {
+                throw "no response"
+            }
+
+            let response = responses.removeFirst()
```

Comment:
this kind of confused me at first.
What do you think of naming this method : `withPagedResponses` and have another method `withResponse` that takes a single response argument (or maybe just add some comment to explain it)

Reply:
The method is currently used for returning pages, but the intention is that it can be used for anything, whether a single (consumable) response, or a series of responses for a series of repeated requests. I'll add a comment.

Reply:
Done 3a0702ab13c79c7fee161a91ba6d6e5c5e9a7628